lib/std/time/epoch.zig

```diff
@@ -40,3 +43,179 @@ pub const morphos = amiga;
 pub const brew = gps;
 pub const atsc = gps;
 pub const go = clr;
+
+/// The type that holds the current year, i.e. 2016
+pub const Year = u16;
+
+pub const epoch_year = 1970;
+pub const secs_per_day: u17 = 24 * 60 * 60;
+
+pub fn isLeapYear(year: Year) bool {
+    if (math.comptimeMod(year, 4) != 0)
+        return false;
+    if (math.comptimeMod(year, 100) != 0)
+        return true;
+    return (0 == math.comptimeMod(year, 400));
+}
+
+test "isLeapYear" {
+    try testing.expectEqual(false, isLeapYear(2095));
+    try testing.expectEqual(true, isLeapYear(2096));
+    try testing.expectEqual(false, isLeapYear(2100));
+    try testing.expectEqual(true, isLeapYear(2400));
+}
+
+pub fn getDaysInYear(year: Year) u9 {
+    return if (isLeapYear(year)) 366 else 365;
+}
+
+pub const YearLeapKind = enum(u1) { not_leap, leap };
+
+pub const Month = enum(u4) {
+    jan,
+    feb,
+    mar,
+    apr,
+    may,
+    jun,
+    jul,
+    aug,
+    sep,
+    oct,
+    nov,
+    dec,
+
+    /// return the numeric calendar value for the given month
+    /// i.e. jan=1, feb=2, etc
+    pub fn numeric(self: Month) u4 {
+        return @enumToInt(self) + 1;
+    }
+};
+
+/// Get the number of days in the given month
+pub fn getDaysInMonth(leap_year: YearLeapKind, month: Month) u5 {
+    return switch (month) {
+        .jan => 31,
+        .feb => @as(u5, switch (leap_year) {
+            .leap => 29,
+            .not_leap => 28,
+        }),
+        .mar => 31,
+        .apr => 30,
+        .may => 31,
+        .jun => 30,
+        .jul => 31,
+        .aug => 31,
+        .sep => 30,
+        .oct => 31,
+        .nov => 30,
+        .dec => 31,
+    };
+}
+
+pub const YearAndDay = struct {
+    year: Year,
+    /// The number of days into the year (0 to 365)
+    day: u9,
```

Comment:
ISO 8601 says that day of the year is in range `[1,365 (366)]`, maybe this should be too?

Reply:
Not sure.  I actually can't think of a use case for wanting the day of the year. For the calculation we need it to be 0 based, so if we make it 1 based, we'll just be adding 1 then undoing that operation in `calculateMonthDay`. An alternative is also to create a method that will return this day value with 1 added. Not sure what the best route is here.